src/topology/instances/add_circle.lean

```diff
@@ -280,6 +285,61 @@ begin
     simpa [zero_add] using (equiv_Ico p 0 u).2.2, },
 end
 
+lemma add_order_of_eq_pos_iff {u : add_circle p} {n : ℕ} (h : 0 < n) :
+  add_order_of u = n ↔ ∃ m < n, gcd m n = 1 ∧ ↑(↑m / ↑n * p) = u :=
+begin
+  refine ⟨λ hu, _, _⟩,
+  { rw ← hu at h,
+    obtain ⟨m, h₀, h₁, h₂⟩ := exists_gcd_eq_one_of_is_of_fin_add_order (add_order_of_pos_iff.mp h),
+    refine ⟨m, _, _, _⟩;
+    rwa ← hu, },
+  { rintros ⟨m, h₀, h₁, rfl⟩,
+    exact add_order_of_div_of_gcd_eq_one h h₁, },
+end
+
+variables (p)
+
+/-- The natural bijection between points of order `n` and natural numbers less than and coprime to
+`n`. -/
+def set_add_order_of_equiv {n : ℕ} (hn : 0 < n) :
+  {u : add_circle p | add_order_of u = n} ≃ {m | m < n ∧ gcd m n = 1} :=
+{ to_fun := λ u, by
+  { let h := (add_order_of_eq_pos_iff hn).mp u.property,
+    exact ⟨classical.some h, classical.some (classical.some_spec h),
+      (classical.some_spec (classical.some_spec h)).1⟩, },
+  inv_fun := λ m, ⟨↑((m : 𝕜) / n * p), add_order_of_div_of_gcd_eq_one hn (m.property.2)⟩,
+  left_inv := λ u, subtype.ext
+    (classical.some_spec (classical.some_spec $ (add_order_of_eq_pos_iff hn).mp u.2)).2,
+  right_inv :=
+  begin
+    rintros ⟨m, hm₁, hm₂⟩,
+    let u : {u : add_circle p | add_order_of u = n} :=
+      ⟨↑((m : 𝕜) / n * p), add_order_of_div_of_gcd_eq_one hn hm₂⟩,
+    let h := (add_order_of_eq_pos_iff hn).mp u.property,
+    ext,
+    let m' := classical.some h,
+    change m' = m,
+    obtain ⟨h₁ : m' < n, h₂ : gcd m' n = 1, h₃ : quotient_add_group.mk ((m' : 𝕜) / n * p) =
+      quotient_add_group.mk ((m : 𝕜) / n * p)⟩ := classical.some_spec h,
+    replace h₃ := congr_arg (coe : Ico 0 (0 + p) → 𝕜) (congr_arg (equiv_Ico p 0) h₃),
+    simpa only [coe_equiv_Ico_mk_apply, mul_left_inj' hp.out.ne', mul_div_cancel _ hp.out.ne',
+      int.fract_div_nat_cast_eq_div_nat_cast_mod,
+      div_left_inj' (nat.cast_ne_zero.mpr hn.ne' : (n : 𝕜) ≠ 0), nat.cast_inj,
+      (nat.mod_eq_iff_lt hn.ne').mpr hm₁, (nat.mod_eq_iff_lt hn.ne').mpr h₁] using h₃,
+  end }
+
+@[simp] lemma card_add_order_of_eq_totient {n : ℕ} (hn : 0 < n) :
```

Comment:
Isn't this one also true for `n = 0`?

Reply:
Actually, the cardinality of order 0 (read: infinity) elements in any monoid is either zero or infinite, and in both case `nat.card` is 0.

Reply:
OK I have added a commit which removes this hypothesis. Thanks for the suggestion!